test/NuGet.Core.Tests/NuGet.Packaging.Test/NuspecReaderTests.cs

```diff
@@ -255,6 +257,83 @@ public class NuspecReaderTests
                   </metadata>
                 </package>";
 
+        private const string VersionRangeInDependency = @"<?xml version=""1.0"" encoding=""utf-8"" standalone=""yes""?>
+                <package xmlns=""http://schemas.microsoft.com/packaging/2011/10/nuspec.xsd"">
+                  <metadata>
+                    <id>PackageA</id>
+                    <version>2.0.1.0</version>
+                    <title>Package A</title>
+                    <authors>ownera, ownerb</authors>
+                    <description>Package A description</description>
+                    <tags>ServiceStack Serilog</tags>
+                    <dependencies>
+                      <dependency id=""PackageB"" version=""{0}"" />
+                    </dependencies>
+                  </metadata>
+                </package>";
+
+        public static IEnumerable<object[]> GetValidVersions()
+        {
+            return GetVersionRange(validVersions: true);
+        }
+
+        public static IEnumerable<object[]> GetInValidVersions()
+        {
+            return GetVersionRange(validVersions: false);
+        }
+
+        private static IEnumerable<object[]> GetVersionRange(bool validVersions)
+        {
+            IEnumerable<string> range = validVersions
+                ? ValidVersionRange()
+                : InvalidVersionRange();
+
+            foreach (var s in range)
+            {
+                yield return new object[] { s };
+            }
+        }
+
+        private static IEnumerable<string> ValidVersionRange()
+        {
+            yield return null;
+            yield return "0.0.0";
+            yield return "1.0.0";
+            yield return "0.0.1";
+            yield return "1.0.0-BETA";
+            yield return "1.0.0";
+            yield return "1.0";
+            yield return "1.0.0.0";
+            yield return "1.0.1";
+            yield return "1.0.01";
+            yield return "00000001.000000000.0000000001";
+            yield return "00000001.000000000.0000000001-beta";
+            yield return "1.0.01-alpha";
+            yield return "1.0.1-alpha.1.2.3";
+            yield return "1.0.1-alpha.1.2.3+metadata";
+            yield return "1.0.1-alpha.1.2.3+a.b.c.d";
+            yield return "1.0.1+metadata";
+            yield return "1.0.1+security.fix.ce38429";
+            yield return "1.0.1-alpha.10.a";
+            yield return "1.0.1--";
+            yield return "1.0.1-a.really.long.version.release.label";
+            yield return "1238234.198231.2924324.2343432";
+            yield return "1238234.198231.2924324.2343432+final";
+            yield return "00.00.00.00-alpha";
+            yield return "0.0-alpha.1";
+            yield return "9.9.9-9";
```

Comment:
Some examples with `[` and `)` and floating would be good. This looks like a list of just versions (not version ranges).

Also, you don't need this many test cases. You can just cover the major different sorts of formats and leave the big test list to `NuGet.Versioning` unit tests.